datafusion/expr/src/udf.rs

```diff
@@ -489,19 +493,40 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
 
     /// Invoke the function with `args` and the number of rows,
     /// returning the appropriate result.
+    ///
+    /// The function will be invoked with the slice of [`ColumnarValue`]
+    /// (either scalar or array).
+    ///
+    /// # Performance
+    ///
+    /// For the best performance, the implementations of `invoke` should handle
```

Comment:
should we refer to `invoke` method anymore as its deprecated?

Reply:
good catch! fixed in https://github.com/apache/datafusion/pull/13174